src/device-manager/WeaveDeviceManager.cpp

```diff
@@ -2726,6 +2741,69 @@ void WeaveDeviceManager::ClearOpState()
     mOpState = kOpState_Idle;
 }
 
+#if WEAVE_CONFIG_DEVICE_MGR_DEMAND_ENABLE_UDP
+
+uint32_t WeaveDeviceManager::sUDPDemandEnableCount = 0;
+
+WEAVE_ERROR WeaveDeviceManager::EnableUDP(void)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+
+    // If UDP listening is NOT already enable in the Weave Message Layer,
+    // enable it now.  Track the number of times UDP is automatically enabled
+    // across all WeaveDeviceManager instances so that it can be automatically
+    // disabled when no longer needed.
+    if (!mMessageLayer->UDPListenEnabled())
+    {
+        WeaveLogProgress(DeviceManager, "Enabling UDP listen");
+        mMessageLayer->SetUDPListenEnabled(true);
+        err = mMessageLayer->RefreshEndpoints();
+        SuccessOrExit(err);
+
+        if (!mUDPEnabled)
+        {
+            sUDPDemandEnableCount++;
+            mUDPEnabled = true;
+        }
+    }
+    else
+    {
+        if (sUDPDemandEnableCount > 0 && !mUDPEnabled)
```

Comment:
Can sUDPDemandEnableCount == 0 in this else block?

Reply:
Yes.  If UDP has been enabled in the Message Layer globally, i.e. before any Device Manager object attempts to use UDP, then sUDPDemandEnableCount will be 0.  This signals to the code that "enable counting" is not being performed because UDP is on globally.